specs/interop/token-bridging.md

```diff
@@ -2,96 +2,296 @@
 
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-**Table of Contents**
 
-- [Interface](#interface)
+- [Overview](#overview)
 - [Implementation](#implementation)
-- [L1 Native Tokens](#l1-native-tokens)
-- [Cross Chain `transferFrom`](#cross-chain-transferfrom)
-- [Factory](#factory)
-- [Upgrading Existing `OptimismMintableERC20Token`s](#upgrading-existing-optimismmintableerc20tokens)
+- [Invariants](#invariants)
+- [Deployment and migrations](#deployment-and-migrations)
+  - [Factory](#factory)
+  - [Migration to the Standard](#migration-to-the-standard)
+- [Future Considerations](#future-considerations)
+  - [Cross Chain `transferFrom`](#cross-chain-transferfrom)
+    - [Approvals](#approvals)
+    - [`remoteTransferFrom()`](#remotetransferfrom)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-Unified ERC20 token bridging built on top of the messaging protocol
-can enable tokens to be fungible and movable across the superchain.
-An issue with bridging is that if the security model is not standardized,
-then the bridged assets are not fungible with each other depending on
-which bridge solution was used to facilitate the transfer between
-domains.
+## Overview
 
-## Interface
+Without a standardized security model, bridged assets may not be fungible with each other. The `SuperchainERC20` unifies ERC20 token bridging to make it fungible across the Superchain. It builds on top of the messaging protocol, as the most trust minimized bridging solution.
 
-The `ISuperchainERC20` interface creates a standard way to send tokens
-between domains. In contrast to [xERC20][xerc20] where the user calls
-a bridge that has `mint` and `burn` privileges on the cross chain enabled
-tokens, the user instead calls methods on the token contract directly.
+Unlike other standards, such as [xERC20](https://www.xerc20.com/), where users interact with a bridge possessing mint and burn privileges on cross-chain enabled tokens, this approach allows users to call methods directly on the token contract.
 
-[xerc20]: https://www.xerc20.com/
+## Implementation
 
-```solidity
-interface ISuperchainERC20 {
-  function bridgeERC20(uint256 amount, uint256 chainId) external;
-  function bridgeERC20To(address to, uint256 amount, uint256 chainId) external;
-  function finalizeBridgeERC20(address to, uint256 amount) external;
-}
-```
+The standard will build on top of ERC20 and include the following three external functions:
 
-## Implementation
+- `sendERC20(uint256 amount, uint256 chainId)(bool)`
+- `sendERC20To(address to, uint256 amount, uint256 chainId)(bool)`
+- `sendERC20To(address to, uint256 amount, uint256 chainId, bytes data)(bool)`
+- `finalizeSendERC20(address to, uint256 amount)`
```

Comment:
What is the rationale for the return values? Three return a bool but the last one doesn't. Should all they return nothing and just revert on failure? 

Also, do we need different function names and arg ordering?  What do you think about:
```solidity
sendERC20(uint256 amount, uint256 chainId)
sendERC20(uint256 amount, uint256 chainId, address to)
sendERC20(uint256 amount, uint256 chainId, address to, bytes data)
```

Reply:
yes, it should all return bool as per the ERC20 standard or not. I was waiting for Mark's answer to change all together. Do you have any take here? Mark suggested to return nothing in a different comment.

 like your suggestion to have a single function name! Will change it :)

Reply:
Yea I agree with @tynes—the `sendERC20` methods don't need to match ERC20, and "the call was successful unless it reverts" is a simpler and common pattern

Reply:
done